src/ForeignKeyDelegate.cpp

```diff
@@ -0,0 +1,156 @@
+#include "sqlitedb.h"
+#include "ForeignKeyDelegate.h"
+
+#include <QRegularExpressionMatch>
+#include <QComboBox>
+#include <QLineEdit>
+#include <QPushButton>
+#include <QHBoxLayout>
+
+class ForeignKeyEditor : public QWidget
+{
+    Q_OBJECT
+
+public:
+    ForeignKeyEditor(QWidget* parent = Q_NULLPTR)
+        : QWidget(parent)
+        , tablesComboBox(new QComboBox(this))
+        , idsComboBox(new QComboBox(this))
+        , m_btnReset(new QPushButton("&Reset", this))
+        , m_clauseEdit(new QLineEdit(this))
+    {
+        m_clauseEdit->setPlaceholderText("foreign key clauses(ON UPDATE, ON DELETE etc.)");
```

Comment:
Hmmm, I'll have to see it in context to decide.

Just added another ToDo item for tomorrow. :wink: